bikesharingweb/pages/signin.js

```diff
@@ -17,62 +13,41 @@ export default class Signin extends Component {
     constructor(props) {
         super(props);
         this.state = {
-            username: '',
-            password: ''
+            users: []
         };
-        const cookies = new Cookies();
-        const user = cookies.get('user');
-        if (user) {
-            console.log('current userId: ' + user.id);
-        } else {
-            console.log('not logged in');
-        }
-
-        this.handleUsername = this.handleUsername.bind(this);
-        this.handlePassword = this.handlePassword.bind(this);
     }
 
     async componentDidMount() {
+        // Clears any login information the user may still have.
+        helpers.logoutUser();
+
+        // Retrieves all users that can be selected for sign-in.
         this.apiHost = await helpers.getApiHostAsync();
-    }
+        const usersResponse = await fetch(`${this.apiHost}/api/user/allUsers`);
+        let users = await usersResponse.json();
+        console.log("Users retrieved", users);
 
-    handleUsername(event) { 
-        this.setState({username: event.target.value});
-    }
+        // Filtering out vendors, as we don't provide any vendors experience for now.
+        users = users.filter(user => user.type != "vendor");
 
-    handlePassword(event) { 
-        this.setState({password: event.target.value });
+        this.setState({users: users});
     }
-    
+
     async handleClick(context) {
-        // Sign in
-        console.log("signing in...");
+        const userId = arguments[0];
+        const userName = arguments[1];
+
+        console.log(`User selected: ${userName} - ${userId}`);
 
-        var url = this.apiHost + '/api/user/auth';
-        
-        const res = await fetch(url, {
-            method: 'POST',
-            cache: 'no-cache',
-            headers: {
-                "Content-Type": "application/json; charset=utf-8"
-            },
-            body: JSON.stringify({
-                username: this.state.username,
-                password: this.state.password
-            }),
-           }  
-        )
-        
-        // get user id
-        const user = await res.json();
-        console.log(user);
-        
         const cookies = new Cookies();
-        cookies.set('user', user);
-        // TODO: authorization confirmation
-        
-        // navigate to index
-        Router.push("/index");
+        cookies.set('user', {
```

Comment:
>cookies.set [](start = 8, length = 11)

Since we have a `helpers.logoutUser()` that clears the cookie, I think it makes sense to move this logic to `helpers.loginUser()` that sets the cookie.

Reply:
Good point, I centralized the cookie logic into helpers.js. That said, I wasn't so happy with the "login"/"logout" naming so I renamed to something more explicit.

---
In reply to: [265360990](https://github.com/ContosoBikeRental/BikeSharingSampleApp/pull/17#discussion_r265360990) [](ancestors = 265360990)